MouseCursor dataset fails integration download test

## Current Behavior
MouseCursor integration test fails during download: https://github.com/aeye-lab/pymovements/actions/runs/14600372728/job/40957033698#step:6:293

https://ars.els-cdn.com/content/image/1-s2.0-S2352340921000160-mmc1.zip seems to be available so I don't understand what the issue is. 

## Expected Behavior
all integration tests pass.

## Minimum acceptance criteria
make MouseCursor integration tests pass

## Context

Some of the code that is called in the test was updated in #1058

I found the regression


first I checked out the commit you mentioned because I also suspected it to be the bad commit-- this worked fine
<details>

<summary> git checkout 70ce9737a0e1ebded2894e53eb30b4e91f30efc1 </summary>

```console
(venv) siqube@c-cube:~/lab/aeye-lab/pymovements
$ git checkout 70ce9737a0e1ebded2894e53eb30b4e91f30efc1
Note: switching to '70ce9737a0e1ebded2894e53eb30b4e91f30efc1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 70ce973 feat: make `mirrors` optional for downloading datasets (#1058)
```

</details>


and ran the integration tests, they pass


<details>

<summary> pytest tests/integration/ -k mousecursor </summary>

```console
(venv) siqube@c-cube:~/lab/aeye-lab/pymovements
$ pytest tests/integration/ -k mousecursor
================================================================================================================= test session starts ==================================================================================================================
platform linux -- Python 3.12.3, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/siqube/lab/aeye-lab/pymovements
configfile: pyproject.toml
collected 48 items / 46 deselected / 2 selected                                                                                                                                                                                                        

tests/integration/download_dataset_test.py .                                                                                                                                                                                                     [ 50%]
tests/integration/public_dataset_processing_test.py .                                                                                                                                                                                            [100%]

=========================================================================================================== 2 passed, 46 deselected in 4.22s ===========================================================================================================

```

</details>

then I scrolled through the logs and found

<details>

<summary> depr: deprecate `utils.downloads` in favor of `Dataset.download()` (#1030) </summary>

```console
(venv) siqube@c-cube:~/lab/aeye-lab/pymovements
$ git checkout 913c372f83d1d19eada650f28fd2ec52d741f0ad
Note: switching to '913c372f83d1d19eada650f28fd2ec52d741f0ad'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 913c372 depr: deprecate `utils.downloads` in favor of `Dataset.download()` (#1030)
```


</details>


where the integration tests fail

<details>

```console
(venv) siqube@c-cube:~/lab/aeye-lab/pymovements
$ pytest tests/integration/download_dataset_test.py::test_download_dataset[MouseCursor]
================================ test session starts ================================
platform linux -- Python 3.12.3, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/siqube/lab/aeye-lab/pymovements
configfile: pyproject.toml
collected 1 item                                                                    

tests/integration/download_dataset_test.py F                                  [100%]

===================================== FAILURES ======================================
________________________ test_download_dataset[MouseCursor] _________________________

truncated...
```

</details>

the commit before passes

<details> 

<summary> Previous HEAD position was 913c372 depr: deprecate `utils.downloads` in favor of `Dataset.download()` (#1030)
HEAD is now at 09b1ee6 ci: remove xfails for #475 as they pass (#1065)</summary>

<details>
(venv) siqube@c-cube:~/lab/aeye-lab/pymovements
$ git checkout 09b1ee6184e346dda52be485dd4922277bab1d58
Previous HEAD position was 913c372 depr: deprecate `utils.downloads` in favor of `Dataset.download()` (#1030)
HEAD is now at 09b1ee6 ci: remove xfails for #475 as they pass (#1065)
</details>

```console
(venv) siqube@c-cube:~/lab/aeye-lab/pymovements
$ pytest tests/integration/download_dataset_test.py::test_download_dataset[MouseCursor]
================================================================================================================= test session starts ==================================================================================================================
platform linux -- Python 3.12.3, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/siqube/lab/aeye-lab/pymovements
configfile: pyproject.toml
collected 1 item                                                                                                                                                                                                                                       

tests/integration/download_dataset_test.py .                                                                                                                                                                                                     [100%]

================================================================================================================== 1 passed in 1.68s ===================================================================================================================
```

</details>

so the problem is in #1030

--- a/src/pymovements/dataset/_utils/_downloads.py
+++ b/src/pymovements/dataset/_utils/_downloads.py
@@ -202,6 +202,9 @@ def _get_redirected_url(url: str, max_hops: int = 3) -> str:
     """
     initial_url = url
     headers = {'Method': 'HEAD', 'User-Agent': USER_AGENT}
+    opener = urllib.request.build_opener()
+    opener.addheaders = [('User-Agent', USER_AGENT)]
+    urllib.request.install_opener(opener)
 
     for _ in range(max_hops + 1):
         with urllib.request.urlopen(urllib.request.Request(url, headers=headers)) as response:

